### DISCUSSION TITLE
Delete a vehicle/device

### DISCUSSION THREAD

**Discussion by SteveDockar** [Hyundai-Kia-Connect/kia_uvo#728] [Q&A]
My search-fu is failing me if this has been asked before, but is it possible to delete a vehicle/device? We have/had two Kia's in the Connect app. We traded one for a newer model. I dropped the old one from Connect and added the new one in its place. The integration added the new vehicle fine, but it didn't do anything about the no longer present older one.

I've disabled the older "device" but would like to remove it if I could, ideally without having to remove the integration and break all the dashboard entries, device_trackers and automations I have using the other two devices. Can anyone help?

Thanks in advance.
Dox

**Comment by cdnninja** [comment]
Have you confirmed it shows no where in the mobile app? I have seen it not delete fully in the app and support had to remove it.

**Reply by SteveDockar** [reply]
As far as I can see there are only two vehicles in the mobile app and the one we replaced has gone. 

I have a second Home Assistant instant which I was feeding the Kia data to over MQTT to save polling the cars twice, but when I installed the integration there just now, it just pulled in the two current vehicles. 

Is it possible the integration is checking for missing vehicles to add but not the other way around?

**Reply by cdnninja** [reply]
It is.  Could you enable debug logging and using  those debug logs confirm no mention of this old car in api responses?

**Reply by SteveDockar** [reply]
Apologies for the delay, I was away over the weekend so didn't get chance to do this until now. I've enabled debug logging, but the only log entries I can see are ones like these:

0-16 17:48:56.084 DEBUG (MainThread) [custom_components.kia_uvo.coordinator] Finished fetching kia_uvo data in 2.020 seconds (success: True)

Is there more to be found somewhere (apologies, I haven't had to do this before)? I see a reference to "Download Diagnostics" in the integration three-dot "more" menu, but I don't see that in this case.

**Comment by fuatakgun** [comment]
please create a request to enable `deletion` functionality. If the car actually exists on the account (not the case for you), it will be created back on next reload/restart.

**Reply by SteveDockar** [reply]
Will do 👍 Thanks for your help with this, and for your efforts with the Integration as a whole, which is excellent!

**Reply by SteveDockar** [reply]
Done see [Issues/763](https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/763). 

Thanks again @fuatakgun !
